### PR TITLE
Feature/forms

### PR DESCRIPTION
**Pull request by errea**
# Description

This PR is requested based on the project on building forms with Rails which include the following:  
- [x] User routes should be only available for new, create, edit, update
- [x] edit actions should find a user based on id from params
- [x] Create should redirect to new user path when user saved
- [x] The update should redirect to edit user if saved successfully
- [x] User params should include username, password, and email
- [x]  form_with uses @user 

# Type of Fix

- [x] Rubocop fix
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

This app has been tested via deployment on Heroku and on the Localhost:3000
- [x] Heroku
- [x] Local Host

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -17,53 +18,41 @@ def new
 
   # GET /users/1/edit
   def edit
+    @user = User.find(params[:id])
   end
 
   # POST /users or /users.json
   def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (GzimAsani) [inline app/controllers/users_controller.rb]: Remove commented out code

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -17,53 +18,41 @@ def new
...
-  def destroy
-    @user.destroy
-    respond_to do |format|
-      format.html { redirect_to users_url, notice: "User was successfully destroyed." }
-      format.json { head :no_content }
+    @user = User.find(params[:id])
+    if @user.update(user_params)
+      flash.notice = "User '#{@user.username}' edited!"
+      redirect_to @user
+      # redirect_to root_path(@user)
```
Comment (GzimAsani) [inline app/controllers/users_controller.rb]: Remove commented out code

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# STATUS: Changes Required :red_circle: :recycle: 

### Hello @errea, your project looks really good :+1: 

### To highlight:
- You have done good job with models/database
- Your project looks professional :+1: 

### To improve:

- [x] Add and comment out forms that use the normal form and form_tag
- [x] It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
- [x] In readme file under installation section add that a user needs to run rails db:migrate to migrate your files
- [x] Your live deployment is returning 404 error. Hint: you are not required to provide live link for this project so either remove or add a live link that works
- [  OPTIONAL ] Add a picture of your project in readme file. It will look more professional 

_Please implement all the above mentioned requirements and ask for another code review whenver you are ready_

### **If you have any question related to the requirements please add a comment here and ping me on slack @GzimAsani**

# HAPPY CODING :computer: :desktop_computer:

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# STATUS: Changes Required :red_circle: 

## Hello @errea,  thank you for implementing almost all the required changes but please also implement the following that was asked previously:

- [x]  _Add and comment out forms that use the normal form and form_tag_

_Please implement all the above mentioned requirements and ask for another code review whenver you are ready_

### **If you have any question related to the requirements please add a comment here and ping me on slack @GzimAsani**

# HAPPY CODING :computer: :desktop_computer:

**Comment by errea** [comment]
@GzimAsani Please I left a message for you on slack.

File: app/models/user.rb [hunk 3]
```diff
@@ -1,2 +1,5 @@
 class User < ApplicationRecord
+  validates :username, presence: true, uniqueness: true
+  validates :email, presence: true, uniqueness: true
+  validates :password, presence: true, length: { minimum: 8 }
```
Comment (uryelah) [inline app/models/user.rb:4]: - [ ] [OPTIONAL] Add some formatting validations to email and password to make sure they are in the expected format, you can use regular expressions for that.

**Review by uryelah — APPROVED** [review]
Hi **Eri Okereafor** , I'm Sarah your reviewer for this PR 👩‍🚀 

Good job adding the fixes required by the last reviewer!


# Approved 🚀 

You may merge your project and submit the completion form. 

------------

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)

**Pull request merged**